src/conf.c

```diff
@@ -535,152 +623,79 @@ conf_set_value_in_file(const char *path, const char *key, const char *value,
 }
 
 bool
-conf_print_items(struct conf *conf,
-                 void (*printer)(const char *descr, const char *origin,
-                                 void *context),
-                 void *context)
+conf_print_value(struct conf *conf, const char *key,
+                 void *context, char **errmsg)
```

Comment:
It's a bit surprising that `context` here is not the same thing as the `context` parameter to `conf_print_items`. Perhaps change from `void *context` to `FILE *file`? And perhaps rename to `conf_print_value_to_file` or introduce a `printer` function parameter like in `conf_print_items`? Doing the latter would open up for writing unit tests without having to print to an actual file.

Reply:
I went for `FILE *file`. I tried first to get a generic `printer` function, but it is not the same signature as for `conf_print_items` and I didn't find an approach that I really liked, so I left this further change for someone else / later.